ingester/ingester.go

```diff
@@ -506,11 +506,15 @@ func (i *Ingester) flushUser(userID string, immediate bool) {
 }
 
 func (i *Ingester) flushSeries(u *userState, fp model.Fingerprint, series *memorySeries, immediate bool) {
-
 	// Enqueue this series flushing if the oldest chunk is older than the threshold
 
 	u.fpLocker.Lock(fp)
-	firstTime := series.head().FirstTime()
+	if len(series.chunkDescs) <= 0 {
+		u.fpLocker.Unlock(fp)
```

Comment:
Yeah, I need to go through this whole thing a factor it out a little better.  Locking is starting to get hard to understand.